File: Restapis/UserService/src/main/java/com/example/demo/entities/Patient.java
package com.example.demo.entities;

import jakarta.persistence.*;
import java.sql.Date;

@Entity
@Table(name = "patient")
public class Patient {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer patientId;

    @ManyToOne
    @JoinColumn(name = "uid")
    private User user;

    private String gender;
    private Date dob;
    private Date registrationDate;

    @Column(nullable = false)
    private String emergencyContact;

    private String allergy;
    private String disease;
    private String bloodGroup;

    public Patient() {}

    // getters and setters
    public Integer getPatientId() { return patientId; }
    public void setPatientId(Integer patientId) { this.patientId = patientId; }

    public User getUser() { return user; }
    public void setUser(User user) { this.user = user; }

    public String getGender() { return gender; }
    public void setGender(String gender) { this.gender = gender; }

    public Date getDob() { return dob; }
    public void setDob(Date dob) { this.dob = dob; }

    public Date getRegistrationDate() { return registrationDate; }
    public void setRegistrationDate(Date registrationDate) { this.registrationDate = registrationDate; }

    public String getEmergencyContact() { return emergencyContact; }
    public void setEmergencyContact(String emergencyContact) { this.emergencyContact = emergencyContact; }

    public String getAllergy() { return allergy; }
    public void setAllergy(String allergy) { this.allergy = allergy; }

    public String getDisease() { return disease; }
    public void setDisease(String disease) { this.disease = disease; }

    public String getBloodGroup() { return bloodGroup; }
    public void setBloodGroup(String bloodGroup) { this.bloodGroup = bloodGroup; }
}

File: Restapis/MedicalRecordsService/MedicalRecordsService/Models/Patient.cs
using System;
using System.Collections.Generic;

namespace MedicalRecordsService.Models;

public partial class Patient
{
    public int PatientId { get; set; }

    public int? Uid { get; set; }

    public string? Gender { get; set; }

    public DateOnly? Dob { get; set; }

    public DateOnly? RegistrationDate { get; set; }

    public string EmergencyContact { get; set; } = null!;

    public string? Allergy { get; set; }

    public string? Disease { get; set; }

    public string? BloodGroup { get; set; }

    public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();

    public virtual ICollection<History> Histories { get; set; } = new List<History>();

    public virtual User? UidNavigation { get; set; }
}
